Remove obsolete note about media keys in MacOS (#6631)
qmk/qmk_firmware: docs/keycodes_basic.md

These keycodes are not part of the Keyboard/Keypad usage page. The `SYSTEM_` keycodes are found in the Generic Desktop page, and the rest are located in the Consumer page.

> Note that some of these keycodes are OS-dependent. For example, on MacOS supports the keycodes `KC_MEDIA_FAST_FORWARD` and `KC_MEDIA_REWIND` in addition to `KC_MEDIA_NEXT_TRACK` and `KC_MEDIA_PREV_TRACK`: those extra keycodes skip within the current track when held, but skip the entire track like the latter keycodes when tapped.

Author: Yeah, but I don't have any basis for that assumption so didn't want to imply it.

> Note that some of these keycodes are OS-dependent. For example, on MacOS supports the keycodes `KC_MEDIA_FAST_FORWARD` and `KC_MEDIA_REWIND` in addition to `KC_MEDIA_NEXT_TRACK` and `KC_MEDIA_PREV_TRACK`: those extra keycodes skip within the current track when held, but skip the entire track like the latter keycodes when tapped.

Reviewer: They are definitely at least recognised by the Linux HID driver:
https://github.com/torvalds/linux/blob/master/drivers/hid/hid-input.c#L943-L946

Whether applications actually do anything with them is a different matter.

> Note that some of these keycodes may be OS-dependent. For example, MacOS supports the keycodes `KC_MEDIA_FAST_FORWARD` and `KC_MEDIA_REWIND` in addition to `KC_MEDIA_NEXT_TRACK` and `KC_MEDIA_PREV_TRACK`: those extra keycodes skip within the current track when held, but skip the entire track like the latter keycodes when tapped.

Reviewer: I still think we could do a little wordsmithing here. Also enclosed in an info box to draw attention to it.

```suggestion
?> Some of these keycodes may behave differently depending on the OS. For example, on macOS, the keycodes `KC_MEDIA_FAST_FORWARD`, `KC_MEDIA_REWIND`, `KC_MEDIA_NEXT_TRACK` and `KC_MEDIA_PREV_TRACK` skip within the current track when held, but skip the entire track when tapped.
```

Author: Maybe, but you've changed the meaning: I believe that only the additional keycodes skip within the current track, but your edited version claims that the others do that too.

Author: You might be right about that, in which case you should feel free to accept your own edit and merge it this is otherwise helpful.

Reviewer: Yep, I can confirm both sets behave exactly the same, at least on Catalina.

Author: Great, done - thanks!
